src/main/kotlin/com/openlattice/chronicle/controllers/SurveyController.kt

```diff
@@ -0,0 +1,185 @@
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun createQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @RequestBody questionnaire: Questionnaire
+    ): UUID {
+        ensureWriteAccess(AclKey(studyId))
+        val id = idGenerationService.getNextId()
+        surveysService.createQuestionnaire(studyId, id, questionnaire)
+
+        return id
+    }
+
+    @DeleteMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH]
+    )
+    override fun deleteQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID
+    ): OK {
+        ensureOwnerAccess(AclKey(studyId))
+        surveysService.deleteQuestionnaire(studyId, questionnaireId)
+        return OK("Successfully deleted questionnaire $questionnaireId")
+    }
+
+    @GetMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID
+    ): Questionnaire {
+        return surveysService.getQuestionnaire(studyId, questionnaireId)
+    }
+
+    @PatchMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        consumes = [MediaType.APPLICATION_JSON_VALUE],
+    )
+    override fun updateQuestionnaire(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID,
+        @RequestBody update: QuestionnaireUpdate
+    ): OK {
+        ensureWriteAccess(AclKey(studyId))
+        surveysService.updateQuestionnaire(studyId, questionnaireId, update)
+
+        return OK("Successfully updated questionnaire $questionnaireId")
+    }
+
+    @GetMapping(
+        path = [STUDY_ID_PATH + QUESTIONNAIRE_PATH],
+        produces = [MediaType.APPLICATION_JSON_VALUE]
+    )
+    override fun getStudyQuestionnaires(@PathVariable(STUDY_ID) studyId: UUID): List<Questionnaire> {
+        return surveysService.getStudyQuestionnaires(studyId)
+    }
+
+    @PostMapping(
+        path = [STUDY_ID_PATH + PARTICIPANT_PATH + PARTICIPANT_ID_PATH + QUESTIONNAIRE_PATH + QUESTIONNAIRE_ID_PATH],
+        produces = []
+    )
+    override fun submitQuestionnaireResponses(
+        @PathVariable(STUDY_ID) studyId: UUID,
+        @PathVariable(PARTICIPANT_ID) participantId: String,
+        @PathVariable(QUESTIONNAIRE_ID) questionnaireId: UUID,
+        @RequestBody responses: List<QuestionnaireResponse>
+    ): OK {
+        surveysService.submitQuestionnaireResponses(studyId, participantId, questionnaireId, responses)
+        return OK()
+    }
+
+    override fun getQuestionnaireResponses(
+        studyId: UUID,
+        questionnaireId: UUID,
+        fileType: FileType
+    ): Iterable<Map<String, Any>> {
```

Comment:
Missing necessary spring annotations.

Reply:
Not needed. downloadQuestionnaireResponses() method serves request to get data